servers/rendering/dummy/storage/material_storage.h

```diff
@@ -31,13 +31,33 @@
 #ifndef MATERIAL_STORAGE_DUMMY_H
 #define MATERIAL_STORAGE_DUMMY_H
 
+#include "servers/rendering/shader_compiler.h"
+#include "servers/rendering/shader_language.h"
 #include "servers/rendering/storage/material_storage.h"
 #include "servers/rendering/storage/utilities.h"
 
 namespace RendererDummy {
 
 class MaterialStorage : public RendererMaterialStorage {
+private:
+	static MaterialStorage *singleton;
+
+	struct DummyShader {
+		HashMap<StringName, ShaderLanguage::ShaderNode::Uniform> uniforms;
+	};
+
+	mutable RID_Owner<DummyShader> shader_owner;
+
+	ShaderCompiler dummy_compiler;
+
 public:
+	static MaterialStorage *get_singleton() {
+		return singleton;
+	};
```

Comment:
```suggestion
	}
```

Reply:
Hmmm, this is just copy-pasted from MeshStorage. It seems that all the Dummy rendering servers use this style. I can just fix all three at once I guess